src/Microsoft.Azure.WebJobs.Host/InvokeFunctionFilterAttribute.cs

```diff
@@ -0,0 +1,122 @@
+        public string ExecutedFilter { get; }
+
+        /// <summary>
+        /// Call the requested job method before the main function call
+        /// </summary>
+        /// <returns></returns>
+        public override async Task OnExecutingAsync(FunctionExecutingContext executingContext, CancellationToken cancellationToken)
+        {
+            if (executingContext == null)
+            {
+                throw new ArgumentNullException("executingContext");
+            }
+            
+            if (!string.IsNullOrEmpty(ExecutingFilter))
+            {
+                executingContext.Logger.LogInformation("Executing Function Filter '" + ExecutingFilter + "'");
+
+                try
+                {
+                    await InvokeJobFunctionAsync(executingContext, cancellationToken);
+                }
+                catch (Exception e)
+                {
+                    executingContext.Logger.LogInformation(e.ToDetails());
+                }
+            }
+            else
+            {
+                await base.OnExecutingAsync(executingContext, cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// Call the requested job method after the main function call
+        /// </summary>
+        /// <returns></returns>
+        public override async Task OnExecutedAsync(FunctionExecutedContext executedContext, CancellationToken cancellationToken)
+        {
+            if (executedContext == null)
+            {
+                throw new ArgumentNullException("executingContext");
+            }
+
+            if (!string.IsNullOrEmpty(ExecutedFilter))
+            {
+                executedContext.Logger.LogInformation("Executing Function Filter '" + ExecutedFilter + "'");
+
+                try
+                {
+                    await InvokeJobFunctionAsync(executedContext, cancellationToken);
+                }
+                catch (Exception e)
+                {
+                    executedContext.Logger.LogInformation(e.ToDetails());
+                }
+            }
+            else
+            {
+                await base.OnExecutedAsync(executedContext, cancellationToken);
+            }
+        }
+
+        internal async Task InvokeJobFunctionAsync<TContext>(TContext context, CancellationToken cancellationToken) where TContext : FunctionInvocationContext
+        {
+            MethodInfo methodInfo = null;
+
+            foreach (var type in context.Config.TypeLocator.GetTypes())
+            {
+                methodInfo = type.GetMethod(ExecutingFilter);
+                if (methodInfo != null)
+                {
+                    break;
+                }
+            }
+
+            IDictionary<string, object> invokeArguments = new Dictionary<string, object>();
+            string parameterName = methodInfo.GetParameters().SingleOrDefault(p => p.ParameterType == typeof(TContext)).Name;
+            invokeArguments.Add(parameterName, context);
```

Comment:
>invokeArguments.Add(parameterName, context); [](start = 12, length = 44)

need a null check on parameterNAme. The target function may not have  a context.